autoload/clap/provider/help_tags.vim

```diff
@@ -0,0 +1,67 @@
+" Author: Mark Wu <markplace@gmail.com>
+" Description: List the help tags.
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+" Help tags support, ported from https://github.com/zeero/vim-ctrlp-help
+let s:help_tags_filecache = get(g:, 'clap_provider_help_tags_cache', fnamemodify($MYVIMRC, ':p:h') . '/clap_provider_help_tags_cache')
+let s:help_tags_memcache = []
+
+function! s:help_tags_source() abort
+  if empty(s:help_tags_memcache)
+    if getftime(s:help_tags_filecache) > max(map(s:get_tags_files(), 'getftime(v:val)'))
+      let s:help_tags_memcache = readfile(s:help_tags_filecache)
+    else
+      let s:help_tags_memcache = s:get_tags_list()
+      silent! call writefile(s:help_tags_memcache, s:help_tags_filecache)
+    endif
+  endif
+
+  return s:help_tags_memcache
+endfunction
+
+function! s:get_tags_list() abort
+  let tagsfiles = s:get_tags_files()
+
+  let input_dict = {}
+  for tagsfile in tagsfiles
+    for line in readfile(tagsfile)
+      let tag = split(line, "\t")
+      let tag_subject = matchstr(line, '^[^\t]\+')
+      if !has_key(input_dict, tag_subject)
+        let input_dict[tag_subject] = printf("%-60s\t%s", tag[0], tag[1])
+      endif
+    endfor
+  endfor
+  call filter(input_dict, 'v:key !~? "^!_TAG_"')
```

Comment:
I'm not sure if this `filter` is neccessary.

Reply:
I am not so sure. For ctags, it will generate meta data start with `!_TAG_XXX`. If vim `helptags` won't generate meta data, I think we can remove it.

** meta data example

```
!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
!_TAG_FILE_SORTED	1	/0=unsorted, 1=sorted, 2=foldcase/
!_TAG_OUTPUT_FILESEP	slash	/slash or backslash/
!_TAG_OUTPUT_MODE	u-ctags	/u-ctags or e-ctags/
!_TAG_PROGRAM_AUTHOR	Universal Ctags Team	//
!_TAG_PROGRAM_NAME	Universal Ctags	/Derived from Exuberant Ctags/
!_TAG_PROGRAM_URL	https://ctags.io/	/official site/
!_TAG_PROGRAM_VERSION	0.0.0	/fefbad26/

```

Reply:
This provider is for help tags exclusively, so we don't have to care about ctags stuff.

Reply:
Done, removed.